### PR TITLE
clean up the artifical viscosity routines

### PR DESCRIPTION
**Pull request by zingale**
we just need to compute divu on the face once
this is prep for limiting the artifical viscosity coefficient for species conservation

## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

**Comment by zingale** [comment]
this is prep for #2228

**Comment by zingale** [comment]
quick tests pass: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2024-07-28/index.html

File: Source/hydro/advection_util.cpp [hunk 1]
```diff
@@ -361,35 +364,38 @@ Castro::apply_av_rad(const Box& bx,
...
+      }
 
-    } else if (idir == 1) {
+      div1 = diff_coeff * std::min(0.0_rt, div1);
 
-      div1 = 0.25_rt * (div(i,j,k) + div(i+1,j,k) +
-                        div(i,j,k+dg2) + div(i+1,j,k+dg2));
-      div1 = diff_coeff * amrex::min(0.0_rt, div1);
-      div1 = div1 * (Erin(i,j,k,n) - Erin(i,j-dg1,k,n));
+      for (int n = 0; n < Radiation::nGroups; ++n) {
```
Comment (maxpkatz) [inline Source/hydro/advection_util.cpp:385]: I don't think this will capture correctly in CUDA.

Comment (zingale) [reply Source/hydro/advection_util.cpp]: fixed

**Review by maxpkatz — APPROVED** [review]

**Pull request merged**